src/core/Pattern_match.ml

```diff
@@ -90,6 +90,7 @@ type t = {
   engine_kind : Engine_kind.t; [@equal fun _a _b -> true]
   (* location info *)
   file : Fpath.t;
```

Comment:
Not clear then why we need 2 fields.

Reply:
Relying on changes in fc35371b4020d3fdf1beb37d3446dbe5f1e81f70 I have refactored this in 9fc64e570.